src/main/index.ts

```diff
@@ -34,126 +35,132 @@ let clusterManager: ClusterManager;
-  } catch (error) {
-    dialog.showErrorBox("Lens Error", `Could not load extensions${error?.message ? `: ${error.message}` : ""}`);
-    console.error(error);
-    console.trace();
-  }
-
-  setTimeout(() => {
-    appEventBus.emit({ name: "service", action: "start" });
-  }, 1000);
-});
-
-app.on("activate", (event, hasVisibleWindows) => {
-  logger.info("APP:ACTIVATE", { hasVisibleWindows });
-
-  if (!hasVisibleWindows) {
-    windowManager?.initMainWindow(false);
-  }
-});
-
-// Quit app on Cmd+Q (MacOS)
-app.on("will-quit", (event) => {
-  logger.info("APP:QUIT");
-  appEventBus.emit({name: "app", action: "close"});
-  event.preventDefault(); // prevent app's default shutdown (e.g. required for telemetry, etc.)
-  clusterManager?.stop(); // close cluster connections
-
-  return; // skip exit to make tray work, to quit go to app's global menu or tray's menu
-});
+      proxyServer = LensProxy.create(proxyPort, clusterManager);
+    } catch (error) {
+      logger.error(`Could not start proxy (127.0.0:${proxyPort}): ${error?.message}`);
+      dialog.showErrorBox("Lens Error", `Could not start proxy (127.0.0:${proxyPort}): ${error?.message || "unknown error"}`);
+      app.exit();
+    }
+
+    extensionLoader.init();
+    extensionDiscovery.init();
+    windowManager = WindowManager.getInstance<WindowManager>(proxyPort);
+
+    // call after windowManager to see splash earlier
+    try {
+      const extensions = await extensionDiscovery.load();
+
+      // Start watching after bundled extensions are loaded
+      extensionDiscovery.watchExtensions();
+
+      // Subscribe to extensions that are copied or deleted to/from the extensions folder
+      extensionDiscovery.events.on("add", (extension: InstalledExtension) => {
+        extensionLoader.addExtension(extension);
+      });
+      extensionDiscovery.events.on("remove", (lensExtensionId: LensExtensionId) => {
+        extensionLoader.removeExtension(lensExtensionId);
+      });
+
+      extensionLoader.initExtensions(extensions);
+    } catch (error) {
+      dialog.showErrorBox("Lens Error", `Could not load extensions${error?.message ? `: ${error.message}` : ""}`);
+      console.error(error);
+      console.trace();
+    }
+
+    setTimeout(() => {
+      appEventBus.emit({ name: "service", action: "start" });
+    }, 1000);
+  })
+  .on("activate", (event, hasVisibleWindows) => {
+    logger.info("APP:ACTIVATE", { hasVisibleWindows });
+
+    if (!hasVisibleWindows) {
+      windowManager?.initMainWindow(false);
+    }
+  })
+  .on("will-quit", (event) => {
+  // Quit app on Cmd+Q (MacOS)
+    logger.info("APP:QUIT");
+    appEventBus.emit({name: "app", action: "close"});
+    event.preventDefault(); // prevent app's default shutdown (e.g. required for telemetry, etc.)
+    clusterManager?.stop(); // close cluster connections
+
+    return; // skip exit to make tray work, to quit go to app's global menu or tray's menu
+  })
+  .on("open-url", (event, rawUrl) => {
```

Comment:
This kind of change to coding style makes huge diff which is really hard to review. Maybe keep the old style just to make the diff smaller?

Reply:
Sounds good, I can make a small PR later